fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/FoldConstantRule.java

```diff
@@ -51,12 +51,15 @@ public List<ExpressionPatternMatcher<? extends Expression>> buildRules() {
 
     /** evaluate by visitor */
     public static Expression evaluate(Expression expr, ExpressionRewriteContext ctx) {
-        if (ctx.cascadesContext != null
```

Comment:
do not change this function. u could do this by modify `org.apache.doris.nereids.rules.expression.rules.FoldConstantRuleOnBE#buildRules`. for example:
```java
    @Override
    public List<ExpressionPatternMatcher<? extends Expression>> buildRules() {
        return ImmutableList.of(
                root(Expression.class)
                        .whenCtx(ctx -> !ctx.cascadesContext.getConnectContext().getSessionVariable().isDebugSkipFoldConstant())
                        .whenCtx(FoldConstantRuleOnBE::isEnableFoldByBe)
                        .thenApply(FoldConstantRuleOnBE::foldByBE)
        );
    }
```

and 

`org.apache.doris.nereids.rules.expression.rules.FoldConstantRuleOnFE#matches`
```java
    private <E extends Expression> ExpressionPatternMatcher<? extends Expression> matches(
            Class<E> clazz, BiFunction<E, ExpressionRewriteContext, Expression> visitMethod) {
        return matchesType(clazz)
                .whenCtx(ctx -> !ctx.cascadesContext.getConnectContext().getSessionVariable().isDebugSkipFoldConstant())
                .whenCtx(NOT_UNDER_AGG_DISTINCT.as())
                .thenApply(ctx -> visitMethod.apply(ctx.expr, ctx.rewriteContext));
    }
```

Reply:
got it. I will fix